src/rdf_graph/rdf_graph.py

```diff
@@ -99,35 +110,64 @@ def cluster_entities(self, embeddings_path, link_method='average'):
 
         self.entity_cluster_results = output
 
-    def extract_raw_tuples(self):
+    def extract_raw_tuples(self, parallel=False, n_cores=5):
         if len(self.all_texts) == 0:
             self.load_texts()
 
-        parser = StanfordParser()
-        i = 0
+        # temp
+        # self.all_texts = {key: self.all_texts[key] for key in list(self.all_texts)}
+
+        if self.verbose: print("{} documents to parse".format(len(self.all_texts)))
 
         # Loop through articles or article batches:
-        for Id, text in self.all_texts.items():
-            print(i)
-            # Start information extraction
-
-            tuples = []
-            sentences = sent_tokenize(text['text'])
-
-            for sentence in sentences:
-                try:
-                    sentence = stanford_parse(parser, sentence)
-                except:
-                    print(sentence)
-                    continue
-                rdfs = [(' '.join(rel['subject']),
-                         ' '.join(rel['relation']),
-                         ' '.join(rel['object'])) for rel in
-                         sentence.get_rdfs(use_tokens=False, how='list')]
-                tuples.append(rdfs)
+        # for Id, text in self.all_texts.items():
+        if parallel:
+            if self.verbose: print("Processing in parallel.")
+            pool = mp.Pool(n_cores)
+            all_tuples = pool.map(self.extract_article_tuples, self.all_texts.values())
+        else:
```

Comment:
So FYI, I was able to track down why this wasn't working. Essentially, because `multiprocessing` uses `pickle` internally to communicate with threads, anything that can't be `pickle`d can't be multiprocessed. That includes the `StanfordParser` object we use in the `extract_article_tuples()` function. To enable parallel execution, we'll need to move that lower - likely make it a part of StanfordParser itself. Something for a future version, I think.

Reply:
(Which is to say, we can lose this conditional here and the parameters for parallel execution in the function definition, but it's not urgent).